### PR TITLE
Discover services

### PR DESCRIPTION
**Pull request by ishit**

### CODE REVIEW

File: app/src/main/java/com/example/android/nsdchat/DiscoverActivity.java [hunk 1]
```diff
@@ -9,26 +9,28 @@
...
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_discover);
 
-        Intent i = getIntent();
-        String[] servicesArray = i.getStringArrayExtra("servicesArray");
-        servicesAdapter = new ArrayAdapter<String>(this, android.R.layout.simple_list_item_1, servicesArray);
-
-        ListView serviceList = (ListView) findViewById(R.id.serviceList);
+        new RefreshList().execute();
+        String[] servicesArray = {};
```
Comment (mirajp1) [inline app/src/main/java/com/example/android/nsdchat/DiscoverActivity.java]: Why is this needed? This array is never updated

File: app/src/main/java/com/example/android/nsdchat/NsdHelper.java [hunk 2]
```diff
@@ -43,7 +43,7 @@ public void initializeNsd() {
     }
 
     public List<String> getServices(){
```
Comment (mirajp1) [inline app/src/main/java/com/example/android/nsdchat/NsdHelper.java:54]: redundant method??

Comment (ishit) [inline app/src/main/java/com/example/android/nsdchat/NsdHelper.java:54]: Yep. Not needed anymore.

**Comment by mirajp1** [comment]
review and merge. Made NsdHelper singleton
